### PR TITLE
Refactor gamemode code

### PR DESCRIPTION
**Pull request by psykzz**
## About The Pull Request

Refactor gamemode code

## Why It's Good For The Game

The idea here being to make it easier to add new gamemodes.

### CODE REVIEW

File: code/game/gamemodes/distress.dm [hunk 1]
```diff
@@ -573,6 +571,45 @@
 
 	to_chat(world, dat)
 
+/datum/game_mode/distress/override_new_player_panel(mob/new_player/NP)
```
Comment (DominikPanic) [inline code/game/gamemodes/distress.dm]: This is the default, not sure if `override` is the best use here.

File: code/game/gamemodes/game_mode.dm [hunk 2]
```diff
@@ -66,10 +77,66 @@
...
+			SSticker.minds += P.new_character.mind
+		CHECK_TICK
+
+
+/datum/game_mode/proc/equip_characters()
+	var/captainless = TRUE
+	for(var/mob/new_player/N in GLOB.player_list)
+		var/mob/living/carbon/human/player = N.new_character
+		if(istype(player) && player.mind && player.mind.assigned_role)
+			if(player.mind.assigned_role == "Captain")
```
Comment (DominikPanic) [inline code/game/gamemodes/game_mode.dm]: We don't have captains here

File: code/game/gamemodes/game_mode.dm [hunk 3]
```diff
@@ -66,10 +77,66 @@
...
+	if(captainless)
+		for(var/mob/new_player/N in GLOB.player_list)
+			if(N.new_character)
+				to_chat(N, "Marine Captain position not forced on anyone.")
+			CHECK_TICK
+
+
+/datum/game_mode/proc/transfer_characters()
+	var/list/livings = list()
+	for(var/mob/new_player/player in GLOB.mob_list)
```
Comment (DominikPanic) [inline code/game/gamemodes/game_mode.dm]: These glob lists could be improved

File: code/game/objects/items/storage/pouch.dm [hunk 4]
```diff
@@ -226,6 +226,14 @@
 	fill_type = /obj/item/ammo_magazine/pistol/vp78
 	fill_number = 3
 
+/obj/item/storage/pouch/magazine/smg/mp7
```
Comment (DominikPanic) [inline code/game/objects/items/storage/pouch.dm]: Unrelated changes

File: code/game/objects/items/pinpointer.dm [hunk 5]
```diff
@@ -1,143 +1,112 @@
+#define PINPOINTER_MODE_DISK "disk"
```
Comment (DominikPanic) [inline code/game/objects/items/pinpointer.dm]: Unrelated cahnges

File: code/game/gamemodes/game_mode.dm [hunk 6]
```diff
@@ -66,10 +77,75 @@
...
+			SSticker.minds += P.new_character.mind
+		CHECK_TICK
+
+
+/datum/game_mode/proc/equip_characters()
+	var/captainless = TRUE
+	for(var/i in GLOB.new_player_list)
+		var/mob/new_player/N = i
+		var/mob/living/carbon/human/player = N.new_character
+		if(istype(player) && player.mind && player.mind.assigned_role)
```
Comment (DominikPanic) [inline code/game/gamemodes/game_mode.dm]: Early continue

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 7]
```diff
@@ -431,6 +431,51 @@ to_chat will check for valid clients itself already so no need to double check f
...
+
+	if (!xeno_candidate || !xeno_candidate.client)
+		return FALSE
+
+	if(living_xeno_queen?.incapacitated(TRUE))
+		to_chat(xeno_candidate, "<span class='warning'>Mother is not in a state to receive us.</span>")
+		return FALSE
+
+	if(!isnewplayer(xeno_candidate))
+		if(!DEATHTIME_CHECK(xeno_candidate))
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm]: Combine with the one above

File: code/game/gamemodes/game_mode.dm [hunk 8]
```diff
@@ -66,10 +77,75 @@
 		qdel(query_round_game_mode)
 
 
+/datum/game_mode/proc/new_player_topic(mob/M, href, href_list[])
```
Comment (DominikPanic) [inline code/game/gamemodes/game_mode.dm]:
```suggestion
/datum/game_mode/proc/new_player_topic(mob/new_player/NP, href, list/href_list)
```

File: code/game/gamemodes/game_mode.dm [hunk 9]
```diff
@@ -98,7 +175,8 @@
 			roletext = "Prefer squad over role"
 
 	//Assemble a list of active players without jobbans.
-	for(var/mob/new_player/player in GLOB.player_list)
+	for(var/i in GLOB.player_list)
```
Comment (spookydonut) [inline code/game/gamemodes/game_mode.dm]: This is an error
